trio/_core/tests/test_ki.py

```diff
@@ -6,13 +6,12 @@
 import threading
 import contextlib
 import time
+import inspect
 
-from async_generator import (
-    async_generator,
-    yield_,
-    isasyncgenfunction,
-    asynccontextmanager,
-)
+try:
+    from async_generator import yield_, async_generator
+except ImportError:
+    async_generator = yield_ = None
```

Comment:
I'm confused-- if `async_generator` is still declared in test dependencies, why would this import fail?

In any case, I don't understand why we're continuing to use the async_generator API in tests.

Reply:
I did it to make sure that the code works both with and without `async_generator` installed and manually uninstalled it. But the risk of a backslide in the future should be pretty slim so yeah can get rid of the conditional.

But keeping tests that check that `@async_generator` works, i.e. `test_async_generator_agen_protection` I think is quite reasonable - if getting rid of those then kinda might as well get rid of async_generator support entirely.

I suppose I can recreate https://github.com/python-trio/async_generator/blob/master/async_generator/_impl.py#L435-L441 if we want to get rid of the dependency in testing too, feels a bit sketch but if we can assume that the implementation will never change then I guess it's okay.

Reply:
> I did it to make sure that the code works both with and without async_generator

That's fair enough, but then the unit tests should explicitly test both those cases, rather than relying on the grace of some developer to manually uninstall async_generator and test it.

Reply:
right, perfect solution would be to fiddle with `sys.modules` in that case. But feels like it's not worth putting a ton of effort into a perfect solution for something that's on the way out soonish. But this feels good enough for now to get the dependency out for the 99% that don't care for it, and cleanup can be put into a separate PR.